### PR TITLE
Cinnamon 6.2 Updates

### PR DESCRIPTION
**Pull request by SparkyBluefang**
Cinnamon 6.2 package updates. all packages should now be EAPI=8 and python 3.13 compat (except for gnome-extra/cinnamon because of dependencies). A number of packaging hacks have also been up-streamed as pull requests.

Also drop Cinnamon 5.8 now that 6.0 is stable.

---

Please check all the boxes that apply:

- [X] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [X] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [X] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

File: gnome-extra/cinnamon/cinnamon-6.2.3.ebuild [hunk 1]
```diff
@@ -95,8 +94,15 @@ RDEPEND="
 	x11-themes/adwaita-icon-theme
 	x11-themes/gnome-themes-standard
 
+	desktop-portal? (
+		sys-apps/xdg-desktop-portal-gtk
+		sys-apps/xdg-desktop-portal-xapp
+	)
+	gnome-online-accounts? (
+		>=net-libs/gnome-online-accounts-3.18.0:=
+	)
```
Comment (eli-schwartz) [inline gnome-extra/cinnamon/cinnamon-6.2.3.ebuild:103]: The USE flag does nothing but add a dependency. It is the wrong dependency though. The settings module checks if https://github.com/xapp-project/gnome-online-accounts-gtk is installed and dynamically adds an entry.

I think it works better as optfeatures, personally, either way that needs to be packaged first.

**Review by eli-schwartz — CHANGES_REQUESTED** [review]

File: gnome-extra/cinnamon/cinnamon-6.2.3.ebuild (hunk 1 above)
Comment (SparkyBluefang) [reply gnome-extra/cinnamon/cinnamon-6.2.3.ebuild:103]: Yuck. I didn't realize that was referencing a different app entirely.

I don't use GOA, and I'm not exactly jumping at the prospect of maintaining another package (especially one I won't be using), so I'm going to leave this out for now. If there's enough demand for it, we can revisit.

File: x11-wm/muffin/muffin-6.2.0.ebuild [hunk 2]
```diff
@@ -91,13 +141,26 @@ src_configure() {
...
+	if use wayland; then
+		emesonargs+=(
+			$(meson_use video_cards_nvidia egl_device)
+			$(meson_use video_cards_nvidia wayland_eglstream)
+		)
+	fi
+
+	# -Werror=lto-type-mismatch
+	# https://bugs.gentoo.org/933879
+	use wayland && filter-lto
```
[the remark this answers is not in the document]
Comment (SparkyBluefang) [reply x11-wm/muffin/muffin-6.2.0.ebuild:162]: It has now

https://github.com/linuxmint/muffin/issues/698

File: gnome-extra/cinnamon-desktop/cinnamon-desktop-6.2.0.ebuild [hunk 3]
```diff
@@ -50,6 +55,7 @@ src_prepare() {
 src_configure() {
 	local emesonargs=(
 		-Dpnp_ids="${EPREFIX}/usr/share/hwdata/pnp.ids"
+		$(meson_feature systemd)
```
[the remark this answers is not in the document]
Comment (SparkyBluefang) [reply gnome-extra/cinnamon-desktop/cinnamon-desktop-6.2.0.ebuild:58]: Good catch. I generally only look at the diffs between tags, and not necessarily at every commit message.

File: gnome-extra/cinnamon/cinnamon-6.2.3.ebuild (hunk 1 above)
Comment (eli-schwartz) [reply gnome-extra/cinnamon/cinnamon-6.2.3.ebuild]: Yeah I don't use it either. I would just leave it out entirely then. ;)

File: x11-libs/xapp/xapp-2.8.4.ebuild [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+		local xapp_python_targets=()
+
+		get_xapp_python_targets() {
+			xapp_python_targets+=("${EPYTHON}")
+		}
+		python_foreach_impl get_xapp_python_targets
+
+		emesonargs+=(
+			-Dpython_target="$(echo "${xapp_python_targets[@]}" | tr ' ' ,)"
+		)
```
Comment (SparkyBluefang) [inline x11-libs/xapp/xapp-2.8.4.ebuild:103]: I think this is nicer than what was here previously, however I'm not sure if this is the best way to do it. I didn't see any other way to get a CSV of the EPYTHON values for the selected python versions.

File: gnome-extra/cinnamon/files/cinnamon-6.2.0-polkit-agent-on-x11.patch [hunk 5]
```diff
@@ -0,0 +1,75 @@
+From c6402be3b5eb84f53367ebd57b1a38664a240d44 Mon Sep 17 00:00:00 2001
+From: Sparky Bluefang <sparky@bluefang-logic.com>
+Date: Wed, 3 Jul 2024 02:10:18 -0400
+Subject: [PATCH] Use Polkit agent for X11 sessions. Don't crash if polkit
+ doesn't start properly.
```
Comment (SparkyBluefang) [inline gnome-extra/cinnamon/files/cinnamon-6.2.0-polkit-agent-on-x11.patch:5]: Worth noting that according to upstream, this doesn't work with multiple users.
https://github.com/linuxmint/cinnamon/pull/12272#issuecomment-2211964115

I'm not sure if that limitation out-weights cleaning-up the ebuild and removing a consumer of `polkit-gnome`.

It also looks like Fedora is shipping this change (without the error handling).

Comment (eli-schwartz) [reply gnome-extra/cinnamon/files/cinnamon-6.2.0-polkit-agent-on-x11.patch:5]: My gut feeling is it's tolerable to have deprecated software used by one more package, until cinnamon gets their replacement fully implemented.

**Comment by juippis** [comment]
@eli-schwartz didn't you use cinnamon so you could runtime test this? :)

I can't compile muffin-6.0.1-r1 on current ~unstable. It looks like gcc-14 related:
```
../muffin-6.0.1/src/core/prefs.c:2261:32: error: passing argument 1 of 'g_strdupv' from incompatible pointer type [-Wincompatible-pointer-types]
 2261 |           strokes = g_strdupv (bindings);
      |                                ^~~~~~~~
      |                                |
      |                                const gchar ** {aka const char **}
In file included from /usr/include/glib-2.0/glib/gstring.h:37,
                 from /usr/include/glib-2.0/glib/giochannel.h:36,
                 from /usr/include/glib-2.0/glib.h:56,
                 from ../muffin-6.0.1/src/core/prefs.c:31:
/usr/include/glib-2.0/glib/gstrfuncs.h:381:55: note: expected 'gchar **' {aka 'char **'} but argument is of type 'const gchar **' {aka 'const char **'}
  381 | gchar**               g_strdupv        (gchar       **str_array);
      |                                         ~~~~~~~~~~~~~~^~~~~~~~~
```

muffin-6.2.0 compiles fine, but it fails tests:
```
Ok:                 4   
Expected Fail:      0   
Fail:               48  
Unexpected Pass:    0   
Skipped:            0   
Timeout:            0
```
(although looks virtualx/gl related)

Everything else at least builds, but I haven't checked **pkgdiff-mg -b** or **qa-cmp** between old and new versions.

**Comment by eli-schwartz** [comment]
The muffin 6.0.1 issue was https://bugs.gentoo.org/919091 and stablereq https://bugs.gentoo.org/935256 -- seems that amd64 was lagging quite a bit...

I have done amd64 testing and runtime testing so I completed that stablereq myself.

**Comment by eli-schwartz** [comment]
muffin 6.2.0 passes tests for me. Thanks for reminding me about this PR as I had strongly intended to take it for a test drive and... somehow forgot??? All packages built and tested cleanly for me, I'll run the new version for the rest of today and push it tonight.

**Pull request closed without merging**